python/dgl/nn/pytorch/utils.py

```diff
@@ -395,3 +397,113 @@ def forward(self, feat_list):
             alpha = self.att(alpha).squeeze(-1)            # (N, num_layers)
             alpha = th.softmax(alpha, dim=-1)
             return (stacked_feat_list * alpha.unsqueeze(-1)).sum(dim=1)
+
+class LabelPropagation(nn.Module):
+    r"""Label Propagation from `Learning from Labeled and Unlabeled Data with Label
+    Propagation <http://mlg.eng.cam.ac.uk/zoubin/papers/CMU-CALD-02-107.pdf>`__
+
+    .. math::
+    \mathbf{Y}^{(t+1)} = \alpha \cdot \tilde{A} \mathbf{Y}^{(t)} + (1 - \alpha)
+    \mathbf{Y}^{(0)},
+    where unlabeled data is inferred by labeled data via propagation. :math:`\alpha`
+    is a weight parameter for balancing between updated labels and initial labels.
+    :math:`\tilde{A}` denotes the normalized adjancency matrix.
+
+    Parameters
+    ----------
+    k: int
+        The number of propagation steps.
+    alpha : float
+        The :math:`\alpha` coefficient in range [0, 1].
+    norm_type : str, optional
+        The type of normalization applied to the adjacency matrix, must be one of the
+        following choices:
+
+        * ``row``: row-normalized adjacency as :math:`D^{-1}A`
+
+        * ``sym``: symmetric normalized adjacency as :math:`D^{-1/2}AD^{-1/2}`
+
+        Default: 'sym'.
+    multi_label : bool, optional
```

Comment:
My bad. I'm now thinking if we should have this flag as it harms flexibility. Probably it's still better to have two separate arguments `clamp` and `normalize`. What do you think? @rudongyu @jermainewang

Reply:
Sure, i think so. Changed.